hummingbot/connector/derivative/coinflex_perpetual/coinflex_perpetual_derivative.py

```diff
@@ -0,0 +1,1391 @@
+        exec_amt_base = decimal_val_or_none(order_data.get("matchQuantity"))
+        if not exec_amt_base:
+            return
+        if not tracked_order:
+            tracked_order = self.in_flight_orders.get(client_order_id)
+        fill_price = decimal_val_or_none(order_data.get("matchPrice", order_data.get("price")))
+        exec_amt_quote = exec_amt_base * fill_price if exec_amt_base and fill_price else None
+        fee_asset = order_data.get("feeInstrumentId", tracked_order.quote_asset)
+        fee_amount = decimal_val_or_none(order_data.get("fees"))
+        position_side = order_data.get("side", "BUY")
+        position_action = (PositionAction.OPEN
+                           if (tracked_order.trade_type is TradeType.BUY and position_side == "BUY"
+                               or tracked_order.trade_type is TradeType.SELL and position_side == "SELL")
+                           else PositionAction.CLOSE)
+        flat_fees = [] if not fee_amount or fee_amount == s_decimal_0 else [TokenAmount(amount=fee_amount, token=fee_asset)]
+
+        fee = TradeFeeBase.new_perpetual_fee(
+            fee_schema=self.trade_fee_schema(),
+            position_action=position_action,
+            percent_token=fee_asset,
+            flat_fees=flat_fees,
+        )
+        trade_update = TradeUpdate(
+            trading_pair=tracked_order.trading_pair,
+            trade_id=int(order_data["matchId"]),
+            client_order_id=client_order_id,
+            exchange_order_id=str(order_data["orderId"]),
+            fill_timestamp=int(order_data["timestamp"]) * 1e-3,
+            fill_price=fill_price,
+            fill_base_amount=exec_amt_base,
+            fill_quote_amount=exec_amt_quote,
+            fee=fee,
+        )
+        self._client_order_tracker.process_trade_update(trade_update=trade_update)
+
+    async def _update_order_fills_from_trades(self, tracked_order, order_update):
+        """
+        This is intended to be a backup measure to get filled events from order status
+        in case CoinFLEX's user stream events are not working.
+        """
+        fee_collected = False
+        for match_data in order_update["matchIds"]:
+            for trade_id in match_data.keys():
+                trade_data = match_data[trade_id]
+                exec_amt_base = decimal_val_or_none(trade_data.get("matchQuantity"))
+                fill_price = decimal_val_or_none(trade_data.get("matchPrice"))
+                exec_amt_quote = exec_amt_base * fill_price if exec_amt_base and fill_price else None
+                if not fee_collected and len(order_update.get("fees", {})):
+                    fee_collected = True
+                    fee_data = order_update.get("fees")
+                    fee_asset = list(fee_data.keys())[0]
+                    fee_amount = decimal_val_or_none(fee_data[fee_asset])
+                else:
+                    fee_asset = tracked_order.quote_asset
+                    fee_amount = None
```

Comment:
Since the fee information in the order status update does not indicate clearly for which trade or trades it applies, processing the trade from the order status information could lead to inaccurate fee information. The trade information should be requested better from its own endpoint if we want to have the HTTP trade update as backup

Reply:
The trades endpoint unfortunately is far too noisy and doesn't allow querying by order id